titan-compiler/checker.lua

```diff
@@ -0,0 +1,55 @@
+local checker = {}
+
```

Comment:
I think I would prefer something similar to the visitor API from [this typed Lua PR](https://github.com/jvprat/typedlua/blob/0569706b0a68a9adfe9aec1ff7146af619008905/typedlua/tlvisitor.lua). Just add an additional "visit_children" method and define the default traversal using pairs/ipairs instead of dozens of mutually recursive methods.